web/src/message_edit.ts

```diff
@@ -1239,25 +1321,35 @@ function handle_message_move_failure_due_to_time_limit(xhr, handle_confirm, on_h
                 "{messages_not_allowed_to_move, plural, one {# message} other {# messages}} will remain in the current topic.",
         },
         {
-            messages_not_allowed_to_move:
-                xhr.responseJSON.total_messages_in_topic - total_messages_allowed_to_move,
+            messages_not_allowed_to_move: total_messages_in_topic - total_messages_allowed_to_move,
         },
     );
 
     const html_body = render_confirm_moving_messages_modal({
         messages_allowed_to_move_text,
         messages_not_allowed_to_move_text,
     });
-    confirm_dialog.launch({
+    let confirm_dialog_opts: DialogWidgetConfig = {
         html_heading: $t_html({defaultMessage: "Move some messages?"}),
         html_body,
         on_click: handle_confirm,
         loading_spinner: true,
-        on_hide: on_hide_callback,
-    });
+    };
+    if (on_hide_callback) {
+        confirm_dialog_opts = {
+            ...confirm_dialog_opts,
+            on_hide: on_hide_callback,
+        };
+    }
```

Comment:
Simpler idiom for this kind of thing:

```suggestion
        ...(on_hide_callback !== undefined && {on_hide: on_hide_callback}),
    };
```

Reply:
Ah nice, thanks!